src/components/ProductCardVertical.jsx

```diff
@@ -1,25 +1,83 @@
-import { useContext, useState } from "react";
-import { Link } from "react-router-dom";
-import { CartContext } from "../contexts/cart-context";
+import { Link, useNavigate } from "react-router-dom";
+import { useCart } from "../contexts/cart-context";
+import { toast } from "react-toastify";
 import "./ProductCardVertical.css";
+import { useAuth } from "../contexts/auth-context";
+import {
+  addItemToCartApi,
+  addToWishlistApi,
+  decreaseItemFromCartApi,
+} from "../apis";
+import { useState } from "react";
+import DecreaseItemQtyButton from "./Buttons/DecreaseItemQtyButton";
+import IncreaseItemQtyButton from "./Buttons/IncreaseItemQtyButton";
+import AddToCartButton from "./Buttons/AddToCartButton";
+import { useWishlist } from "../contexts/wishlist-context";
 
 export const ProductCardVertical = ({ item, index }) => {
-  const cart = useContext(CartContext);
-  const [itemCartQty, setItemCartQty] = useState(0);
+  const auth = useAuth();
+  const { cartState, cartDispatch, CART_ACTIONS } = useCart();
+  const { wishlist, setWishlist } = useWishlist();
+  const [btnLoading, setBtnLoading] = useState(false);
+  const navigate = useNavigate();
+  let cartQty = 0;
+  for (let cartItem of cartState.cart) {
+    if (cartItem.slug === item.slug) {
+      cartQty = cartItem.cartQty;
+    }
+  }
+  let inWishlist = false;
+  for (let wishlistItem of wishlist) {
+    if (wishlistItem.slug === item.slug) {
+      inWishlist = true;
+    }
+  }
+
+  const itemCartAddLoginToast = (item) => {
+    toast.info(`Please login to add items to your cart`, {
+      theme: "dark",
+      hideProgressBar: true,
+      position: "bottom-center",
+      toastId: "itemCartAddLoginToast",
+    });
+  };
+  const itemWishlistLoginToast = (item) => {
+    toast.info(`Please login to add items to your wishlist`, {
+      theme: "dark",
+      hideProgressBar: true,
+      position: "bottom-center",
+      toastId: "itemWishlistLoginToast",
+    });
+  };
 
   const decCartQty = () => {
```

Comment:
This function is repeated in CartProductCard.jsx as well. You can reuse repeated functions.